### PR TITLE
Ms 188 rework rt api

### PR DESCRIPTION
**Pull request by imda-kelvinkok**

### CODE REVIEW

File: moonshot/integrations/cli/cli.py [hunk 1]
```diff
@@ -41,16 +44,20 @@ def default(self, statement: Statement) -> None:
...
         if active_session:
+            if active_session["context_strategy"]:
+                cs_prompt = f"CS: ({active_session['context_strategy']},{active_session['cs_num_of_prev_prompts']})] > "
+            else:
+                cs_prompt = f"CS: {active_session['context_strategy']}]> "
             self.prompt = (
                 f"moonshot ({active_session['session_id']}) "
                 f"[PT: {active_session['prompt_template']}, "
-                f"CS: {active_session['context_strategy']}] > "
+                f"{cs_prompt}"
```
Comment (imda-lionelteo) [inline moonshot/integrations/cli/cli.py:60]: just checking. Dont need to put the CS: in front or something anymore?

File: moonshot/src/api/api_session.py [hunk 2]
```diff
@@ -57,9 +88,10 @@ def api_get_available_session_info() -> tuple[list, list]:
     runner_with_session_db_list = []
 
     for runner_instance in runner_instances:
-        if Session.load(runner_instance.database_instance) is not None:
-            runner_ids.append(runner_instance.id)
-            runner_with_session_db_list.append(runner_instance.database_instance)
+        if runner_instance.database_instance:
+            if Session.load(runner_instance.database_instance) is not None:
+                runner_ids.append(runner_instance.id)
+                runner_with_session_db_list.append(runner_instance.database_instance)
```
Comment (imda-lionelteo) [inline moonshot/src/api/api_session.py:94]: just curious what is going to happen if the database instance is None or session is None

File: moonshot/src/redteaming/attack/attack_module.py [hunk 3]
```diff
@@ -27,18 +27,19 @@ class AttackModule:
...
-        self.connector_ids = am_args.connector_eps
-        self.prompt_templates = am_args.prompt_templates
-        self.prompt = am_args.prompt
-        self.system_prompt = am_args.system_prompt
-        self.metric_ids = am_args.metric_ids
-        self.context_strategy_info = am_args.context_strategy_info
-        self.db_instance = am_args.db_instance
-        self.params = am_args.params
+    def __init__(self, am_args: AttackModuleArguments | None = None):
+        if am_args:
```
Comment (imda-lionelteo) [inline moonshot/src/redteaming/attack/attack_module.py:31]: Why am_args will be None

File: moonshot/integrations/cli/cli.py (hunk 1 above)
Comment (imda-kelvinkok) [reply moonshot/integrations/cli/cli.py:60]: `cs_prompt` is a variable with the `CS:` appended in front

File: moonshot/src/api/api_session.py (hunk 2 above)
Comment (imda-kelvinkok) [reply moonshot/src/api/api_session.py:94]: if a database instance is None, that particular runner will be skipped.
if a Session is None, that particular runner will be skipped as well. This should not be a None as if there is no Session detected in the runner, one will be created but a None will be returned if there is an error creating the Session.
if all are None, empty lists will be returned

File: moonshot/src/redteaming/attack/attack_module.py (hunk 3 above)
Comment (imda-kelvinkok) [reply moonshot/src/redteaming/attack/attack_module.py:31]: this is complex. will explain more later

**Review by imda-lionelteo — APPROVED** [review]

**Pull request merged**